docs/current/sdk/go/guides/544174-multistage-build.md

```diff
@@ -0,0 +1,37 @@
+---
+slug: /544174/multistage-build
+---
+
+# Multistage Builds
+
+A common practice when building containers with Dockerfiles is something called a multistage build. This means within the `docker build`, your application is compiled in a context which has tools that are required for building your app, but not necessarily required for running the application. To reduce the number of dependencies in the image you actually run the application on, the compiled application is copied to a different base image which only has the required components to run the application. More information about multistage builds can be found in [this guide](https://docs.docker.com/build/building/multi-stage/) from Docker.
+
+This document shows how multistage builds are created using the Go SDK.
+
+## Introduction
+
+## Requirements
+
+This guide assumes that:
+
+- You have a Go development environment with Go 1.15 or later. If not, [download and install Go](https://go.dev/doc/install).
+- You are familiar with the basics of the Go SDK and have it installed. If not, read the [Go SDK guide](../959738-get-started.md) and the [Go SDK installation instructions](../371491-install.md).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+
+## Example
+
+The following code snippet demonstrates a multistage build with the Go SDK. Below the snippet, we'll break down exactly what's happening
+
+```go file=../snippets/multistage-build/main.go
+
+```
+
+The beginning of the file starts by creating a Dagger client and loading the project which is going to be built. Once we have a reference to the project, we build the application by using the `golang:latest` image to mount the source directory, set `CGO_ENABLED=` since the binary will be published on Alpine, and then execute `go build`.
```

Comment:
```suggestion
This code listing starts by creating a Dagger client and loading the project to be built. It obtains a reference to the project and then builds the application by using the `golang:latest` image to mount the source directory, sets `CGO_ENABLED=` since the binary will be published on `alpine`, and executes `go build`.
```